### PR TITLE
chore: fix some easy circular dependencies in dep graph

### PR DESCRIPTION
**Pull request by astone123**
This PR resolves some circular dependencies between the cli package and `@cypress/*` packages, as well as between `@packages/config` and `@packages/errors`

### CODE REVIEW

File: npm/webpack-preprocessor/package.json [hunk 1]
```diff
@@ -34,7 +34,6 @@
     "chalk": "3.0.0",
     "chokidar-cli": "2.1.0",
     "common-tags": "^1.8.2",
-    "cypress": "0.0.0-development",
```
Comment (jordanpowell88) [inline npm/webpack-preprocessor/package.json]: Do we need to remove this?

Comment (astone123) [reply npm/webpack-preprocessor/package.json]: I don't think so... maybe we should keep it? I'm not entirely sure of the consequences

Comment (jordanpowell88) [reply npm/webpack-preprocessor/package.json]: I see the `cypress` types being used still in the lib. So I think we just put this back if it doesn't cause a circular dep

Comment (astone123) [reply npm/webpack-preprocessor/package.json]: yeah I'll add them back

**Pull request merged**